P202-Training.Domain/UsersService.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using AutoMapper;
+using P202.Training.Data;
+using P202.Training.Data.Entities;
+using P202.Training.Data.Repositories;
+using P202.Training.Domain.Models;
+
+namespace P202.Training.Domain
+{
+    public class UsersService : IUsersService
+    {
+        public void CreateUser(Users users)
+        {
+            if (users == null) return;
+            Mapper.Initialize(cfg => cfg.CreateMap<Users, User>());
+            var mapUser = Mapper.Map<User>(users);
+            var userRepository = new UserRepository(new NHibernateSessionManager());
```

Comment:
Same here, you sholdn't be creating a new `UserRepository`, you should inject the `UserRepository` and let the DI engine handle object creation